src/Adapter/Product/Repository/ProductMultiShopRepository.php

```diff
@@ -658,6 +665,260 @@ public function getProductByDefaultShop(ProductId $productId): Product
         return $this->getProductByShopId($productId, $defaultShopId);
     }
 
+    /**
+     * Gets position product position in category
+     *
+     * @param ProductId $productId
+     * @param CategoryId $categoryId
+     *
+     * @return int|null
+     */
+    public function getPositionInCategory(ProductId $productId, CategoryId $categoryId): ?int
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb->select('position')
+            ->from($this->dbPrefix . 'category_product')
+            ->where('id_product = :productId')
+            ->andWhere('id_category = :categoryId')
+            ->setParameter('productId', $productId->getValue())
+            ->setParameter('categoryId', $categoryId->getValue())
+        ;
+
+        $position = $qb->execute()->fetchOne();
+
+        if (!$position) {
+            return null;
+        }
+
+        return (int) $position;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     *
+     * @return array<array<string, string>>
+     *                                      e.g [
+     *                                      ['id_product' => '1', 'name' => 'Product name', 'reference' => 'demo15'],
+     *                                      ['id_product' => '2', 'name' => 'Product name2', 'reference' => 'demo16'],
+     *                                      ]
+     *
+     * @throws CoreException
+     */
+    public function getRelatedProducts(ProductId $productId, LanguageId $languageId): array
+    {
+        $this->assertProductExists($productId);
+        $productIdValue = $productId->getValue();
+
+        try {
+            $accessories = Product::getAccessoriesLight($languageId->getValue(), $productIdValue);
+        } catch (PrestaShopException $e) {
+            throw new CoreException(sprintf(
+                'Error occurred when fetching related products for product #%d',
+                $productIdValue
+            ));
+        }
+
+        return $accessories;
+    }
+
+    /**
+     * @param ProductId $productId
+     *
+     * @throws ProductNotFoundException
+     */
+    public function assertProductExists(ProductId $productId): void
+    {
+        $this->assertObjectModelExists($productId->getValue(), 'product', ProductNotFoundException::class);
+    }
+
+    /**
+     * @param ProductId[] $productIds
+     *
+     * @throws ProductNotFoundException
+     */
+    public function assertAllProductsExists(array $productIds): void
+    {
+        //@todo: no shop association. Should it be checked here?
```

Comment:
ping @jolelievre for consultation :smile: 
this used  where we only check if product exist overall (not considering shops), so either we leave it this way, or we can make it to assertAllProductsExistsInshop and provide default product shop as argument?